gh-492 Display version and branch name of terminology when selecting for Code Sets

pjmonks opened the pull request:
Resolves #492 

![image](https://user-images.githubusercontent.com/3219480/188189180-c1167248-bbad-48c2-9793-89f8522a4086.png)

jamesrwelch commented:
This is brilliant - could we roll this out easily to other places too - for example when we create model data types?  Is there an easy way to find other similar model drop-downs?

pjmonks commented:
@jamesrwelch I've updated other areas which use a similar control:

![image](https://user-images.githubusercontent.com/3219480/188432898-c977d95b-a717-4b6b-9631-21d914611cba.png)

I've updated the New Data Type wizard and the Element Selector (links for description fields).

Now that the label with branch/version is a shared component I've also looked at the tree views too. The new Favourites tree is updated but not the main model tree - too complex to fiddle with at the moment.

jamesrwelch reviewed (approved):
This works really nicely - thank you!
